### ISSUE TITLE
[Feature Request] Re-enable multi-output logs

### ISSUE THREAD

**Issue by innat** [keras-team/keras#19219]
If a model is configured for multi-output, in keras 2, we can get the log message for all corresponding outputs. But in keras 3, it is removed and only shows the summarized log message. Unlike mult-class classification, returning log message for mult-output sounds more natural. 

```python
priority_pred = layers.Dense(1, name="priority")(x)
department_pred = layers.Dense(num_departments, name="department")(x)

model = keras.Model(
    inputs=[title_input, body_input, tags_input],
    outputs={"priority": priority_pred, "department": department_pred},
)

model.compile(
    optimizer=keras.optimizers.RMSprop(1e-3),
    loss={
        "priority": keras.losses.BinaryCrossentropy(from_logits=True),
        "department": keras.losses.CategoricalCrossentropy(from_logits=True),
    },
    loss_weights={"priority": 1.0, "department": 0.2},
)

model.fit
```

in keras 2

```python
Epoch 1/2
 40/40 ━ 1s 12ms/step - loss: ?, priority_loss: ?, department_loss: ?
```

in keras 3

```python
Epoch 1/2
 40/40 ━ 1s 12ms/step - loss: ?
```

Posted as a bug https://github.com/keras-team/keras/issues/19159, but it turned out that it was intentionally removed.

**Comment by james77777778** [comment]
+1 for this

I have also filed the issue at #18993

**Comment by arthurflor23** [comment]
I'm having the same restriction using GAN models, which have several loss values...

**Comment by innat** [comment]
@fchollet 
Just to be clear, in keras 2, we can get 

```python
Epoch 1/2
 40/40 ━ 1s 12ms/step - loss: ?, priority_loss: ?, department_loss: ?
```

but in keras 3

```python
Epoch 1/2
 40/40 ━ 1s 12ms/step - loss: ?
```

So, it's not about the metrics but loss scores of a multi output model. And it seems missing in the log in keras 3. We found this unexpected, coz sometimes we prefer to monitor target loss score using callback to reduce lr or saving checkpoint, etc. As this ticket is closed, I guess this is fixed but I've just tried a sample code with keras-nightly (3.2.0.dev2024040203), it's same as before. 

```python

model.compile(
    optimizer=keras.optimizers.RMSprop(1e-3),
    loss={
        "priority": keras.losses.BinaryCrossentropy(from_logits=True),
        "department": keras.losses.CategoricalCrossentropy(from_logits=True),
    },
    loss_weights={"priority": 0.1, "department": 0.2},
)

Epoch 1/2
40/40 ━━━━━━━━━━━━━━━━━━━━ 5s 48ms/step - loss: 0.6244
Epoch 2/2
40/40 ━━━━━━━━━━━━━━━━━━━━ 2s 44ms/step - loss: 0.6285
```
